// next13-progressbar/dist/index.js
"use strict";var C=Object.create;var g=Object.defineProperty;var M=Object.getOwnPropertyDescriptor;var $=Object.getOwnPropertyNames;var F=Object.getPrototypeOf,T=Object.prototype.hasOwnProperty;var o=(r,e)=>g(r,"name",{value:e,configurable:!0});var B=(r,e)=>{for(var t in e)g(r,t,{get:e[t],enumerable:!0})},k=(r,e,t,c)=>{if(e&&typeof e=="object"||typeof e=="function")for(let s of $(e))!T.call(r,s)&&s!==t&&g(r,s,{get:()=>e[s],enumerable:!(c=M(e,s))||c.enumerable});return r};var P=(r,e,t)=>(t=r!=null?C(F(r)):{},k(e||!r||!r.__esModule?g(t,"default",{value:r,enumerable:!0}):t,r)),K=r=>k(g({},"__esModule",{value:!0}),r);var z={};B(z,{Next13ProgressBar:()=>U,default:()=>_,useRouter:()=>v});module.exports=K(z);var l=P(require("react")),p=P(require("nprogress")),u=require("next/navigation");var N=l.default.memo(({color:r="#0A2FFF",height:e="2px",options:t,showOnShallow:c=!1,delay:s=0,style:h})=>{let m=l.default.createElement("style",null,h||`
          #nprogress {
            pointer-events: none;
          }
          
          #nprogress .bar {
            background: ${r};
          
            position: fixed;
            z-index: 1031;
            top: 0;
            left: 0;
          
            width: 100%;
            height: ${e};
          }
          
          /* Fancy blur effect */
          #nprogress .peg {
            display: block;
            position: absolute;
            right: 0px;
            width: 100px;
            height: 100%;
            box-shadow: 0 0 10px ${r}, 0 0 5px ${r};
            opacity: 1.0;
          
            -webkit-transform: rotate(3deg) translate(0px, -4px);
                -ms-transform: rotate(3deg) translate(0px, -4px);
                    transform: rotate(3deg) translate(0px, -4px);
          }
          
          /* Remove these to get rid of the spinner */
          #nprogress .spinner {
            display: block;
            position: fixed;
            z-index: 1031;
            top: 15px;
            right: 15px;
          }
          
          #nprogress .spinner-icon {
            width: 18px;
            height: 18px;
            box-sizing: border-box;
          
            border: solid 2px transparent;
            border-top-color: ${r};
            border-left-color: ${r};
            border-radius: 50%;
          
            -webkit-animation: nprogress-spinner 400ms linear infinite;
                    animation: nprogress-spinner 400ms linear infinite;
          }
          
          .nprogress-custom-parent {
            overflow: hidden;
            position: relative;
          }
          
          .nprogress-custom-parent #nprogress .spinner,
          .nprogress-custom-parent #nprogress .bar {
            position: absolute;
          }
          
          @-webkit-keyframes nprogress-spinner {
            0%   { -webkit-transform: rotate(0deg); }
            100% { -webkit-transform: rotate(360deg); }
          }
          @keyframes nprogress-spinner {
            0%   { transform: rotate(0deg); }
            100% { transform: rotate(360deg); }
          }
        `);p.default.configure(t||{});let f=(0,u.usePathname)(),y=(0,u.useSearchParams)();return(0,l.useEffect)(()=>{p.default.done()},[f,y]),(0,l.useEffect)(()=>{let d,R=o(()=>{d=setTimeout(p.default.start,s)},"startProgress"),L=o(()=>{clearTimeout(d),p.default.done()},"stopProgress"),E=o(a=>{let i=a.currentTarget;if(i.target==="_blank"||i.target==="_top"||i.target==="_parent"||i.hasAttribute("download"))return;let n=new URL(i.href),b=new URL(location.href),O=n?.searchParams?.toString()!==b?.searchParams?.toString()&&n?.search!==b?.search,S=n?.pathname===b?.pathname&&!O;a.metaKey||a.ctrlKey||a.shiftKey||a.altKey||c&&S||S||R()},"handleAnchorClick"),A=o(()=>{let a=document.querySelectorAll("a");Array.from(a).filter(n=>n.href.startsWith("tel:+")||n.href.startsWith("mailto:")?!1:n.href&&n.target!=="_blank").forEach(n=>n.addEventListener("click",E))},"handleMutation");new MutationObserver(A).observe(document,{childList:!0,subtree:!0}),window.history.pushState=new Proxy(window.history.pushState,{apply:(a,i,n)=>(L(),a.apply(i,n))})},[]),m},()=>!0);function v(){let r=(0,u.useRouter)(),e=(0,u.usePathname)();function t(s,h){let m=new URL(s,location.href),f=new URL(location.href),d=m?.searchParams?.toString()!==f?.searchParams?.toString()&&m?.search!==f?.search;return m.pathname===e&&!d?Promise.resolve(!0):(p.default.start(),r.push(s,h))}o(t,"push");function c(s,h){return new URL(s,location.href).pathname===e?Promise.resolve(!0):(p.default.start(),r.replace(s,h))}return o(c,"replace"),{...r,push:t,replace:c}}o(v,"useRouter");var x=P(require("react"));function w(r){return o(function(t){return x.default.createElement(x.default.Suspense,null,x.default.createElement(r,{...t}))},"WithSuspenseComponent")}o(w,"withSuspense");var U=w(N);var _=U;0&&(module.exports={Next13ProgressBar,useRouter});
